### PR TITLE
Eint comparison functions and unit tests

### PR DESCRIPTION
**Pull request by mikurys**

### CODE REVIEW

File: src/stdplus/eint.hpp [hunk 1]
```diff
@@ -82,6 +82,75 @@ T eint_plus(T a, T b, typename std::enable_if_t<std::is_integral<T>::value>* = 0
 	return a+b;
 }
 
+template <typename Ta, typename Tb>
+bool eint_less(Ta a, Tb b, typename std::enable_if_t<std::is_integral<Ta>::value && std::is_integral<Tb>::value>* = 0
+							, typename std::enable_if_t<std::is_unsigned<Ta>::value>* = 0
+							, typename std::enable_if_t<std::is_signed<Tb>::value>* = 0) {
+	if (b<0) return false;
+	else {
+		if (sizeof(Ta) >= sizeof(Tb)){
```
Comment (rfree-d) [inline src/stdplus/eint.hpp]: maybe better do compare
std::numeric_limits<S>::digits of them
or something like that

File: src/stdplus/eint.hpp [hunk 2]
```diff
@@ -82,6 +82,75 @@ T eint_plus(T a, T b, typename std::enable_if_t<std::is_integral<T>::value>* = 0
...
+
+template <typename Ta, typename Tb>
+bool eint_equal(Ta a, Tb b, typename std::enable_if_t<std::is_integral<Ta>::value && std::is_integral<Tb>::value>* = 0
+							, typename std::enable_if_t<std::is_signed<Ta>::value == std::is_signed<Tb>::value>* = 0) {
+	return a==b;
+}
+
+template <typename Ta, typename Tb>
+bool eint_greater(Ta a, Tb b, typename std::enable_if_t<std::is_integral<Ta>::value && std::is_integral<Tb>::value>* = 0) {
+	return !eint_less(a, b);
```
Comment (rfree-d) [inline src/stdplus/eint.hpp]: can be equall

File: src/stdplus/eint.hpp (hunk 1 above)
Comment (mikurys) [inline src/stdplus/eint.hpp]: done

File: src/stdplus/eint.hpp (hunk 2 above)
Comment (mikurys) [inline src/stdplus/eint.hpp]: fixed

**Comment by rfree-d** [comment]
Opening instead PR #379 with more commits (so we can merge it now), please follow there

**Pull request closed without merging**